index.d.ts

```diff
@@ -1,3 +1,9 @@
 import tape = require('tape');
 
+declare module 'tape' {
+  interface Test {
+    shouldFail(p: PromiseLike<any>, c?: new() => Error): PromiseLike<any>;
```

Comment:
Please import the `es6-promise` library. This interface relies on a global from TypeScript that is **only** available in ES6+.